tutorials/build.gradle.kts

```diff
@@ -0,0 +1,76 @@
+data class SnippetData(
+  val file: File,
+  val lineNumber: Int,
+  val content: String,
+  var tempDir: File? = null
+)
+
+fun findSnippets(dirs: List<String>): List<SnippetData> {
+  val snippets = mutableListOf<SnippetData>()
+  dirs.forEach { dirName ->
+    fileTree(rootProject.projectDir.resolve(dirName)).filter { it.name.endsWith(".md") }.forEach { file ->
+      var currentSnippet = kotlin.text.StringBuilder()
+      var inSnippet = false
+      var lineNumber = 0
+      file.forEachLine { line ->
+        lineNumber++
+        if (line =="```kotlin")
+          inSnippet = true
+        else if (line == "```" && inSnippet) {
+          inSnippet = false
+          snippets.add(SnippetData(file, lineNumber, currentSnippet.toString()))
+          currentSnippet.clear()
+        } else {
+          if (inSnippet) {
+            currentSnippet.appendln(line)
+          }
+        }
+      }
+    }
+  }
+  return snippets
+}
+
+fun cloneTemplate(index: Int, content: String): File {
+  val tempDir = file("${project.buildDir.absolutePath}/temp/cloned-$index")
+  tempDir.deleteRecursively()
+  tempDir.mkdirs()
+  file("${projectDir.parentFile.absolutePath}/templates/desktop-template").copyRecursively(tempDir)
+  //tempDir.deleteOnExit()
+  File("$tempDir/src/main/kotlin/main.kt").printWriter().use { out ->
+    out.println(content)
+  }
+  return tempDir
+}
+
+fun checkDirs(dirs: List<String>) {
+  val snippets = findSnippets(dirs)
+  snippets.forEachIndexed { index, snippet ->
+    println("process snippet $index at ${snippet.file}:${snippet.lineNumber}")
+    snippet.tempDir = cloneTemplate(index, snippet.content)
+    val proc = ProcessBuilder("bash", "./gradlew", "build")
+      .directory(snippet.tempDir)
+      .redirectOutput(ProcessBuilder.Redirect.PIPE)
+      .redirectError(ProcessBuilder.Redirect.PIPE)
+      .start()
+    proc.waitFor(5, TimeUnit.MINUTES)
+    if (proc.exitValue() != 0) {
+      System.err.println("Error in snippet at ${snippet.file}:${snippet.lineNumber}")
+      println(proc.inputStream.bufferedReader().readText())
+      println(proc.errorStream.bufferedReader().readText())
```

Comment:
We need to fail gradle task if some tutorial contains error. So CI can tell us if something wrong.